code/tracing/TraceEventWriter.cpp

```diff
@@ -0,0 +1,111 @@
+#include "tracing/TraceEventWriter.h"
+
+namespace
+{
+using namespace tracing;
+
+const char* getTypeStr(tracing::EventType type) {
+	switch(type) {
+		case tracing::EventType::Complete:
+			return "X";
+		case tracing::EventType::Begin:
+			return "B";
+		case tracing::EventType::End:
+			return "E";
+		case EventType::AsyncBegin:
+			return "b";
+		case EventType::AsyncStep:
+			return "n";
+		case EventType::AsyncEnd:
+			return "e";
+		default: 
+			Assertion(false, "Invalid enum value!");
+			return "";
+	}
+}
+
+void writeTime(std::ofstream& out, std::uint64_t time) {
+	// Save stream state
+	auto flags = out.flags();
+	out << std::fixed;
+
+	out << (time / 1000.);
```

Comment:
The C++ stream format flags are very weird to work with. I guess `setw` should work here so I'll try that.